packages/@aws-cdk/aws-codepipeline/test/pipeline.test.ts

```diff
@@ -389,6 +389,49 @@ describe('', () => {
 
 
       });
+
+      test('does not allow key rotation if cross account keys has not been set', () => {
+        const app = new cdk.App();
+        const stack = new cdk.Stack(app, 'PipelineStack');
+        const sourceOutput = new codepipeline.Artifact();
+
+        expect(() => {
+          new codepipeline.Pipeline(stack, 'Pipeline', {
+            crossAccountKeys: false,
+            enableKeyRotation: true,
+            stages: [
+              {
+                stageName: 'Source',
+                actions: [new FakeSourceAction({ actionName: 'Source', output: sourceOutput })],
+              },
+            ],
+          });
+        }).toThrow("Setting 'enableKeyRotation' to true also requires 'crossAccountKeys' to be enabled");
+      });
+
+      test('allows enabling key rotation if cross account keys has been set', () => {
+        const app = new cdk.App();
+        const stack = new cdk.Stack(app, 'PipelineStack');
+        const sourceOutput = new codepipeline.Artifact();
+        new codepipeline.Pipeline(stack, 'Pipeline', {
+          enableKeyRotation: true,
+          crossAccountKeys: true,
```

Comment:
This line should be left out (it's needed because of the mistake I mentioned above):

```suggestion
```